src/commands/cicdPipeline.ts

```diff
@@ -0,0 +1,53 @@
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import * as utils from 'util';
+import { IActionContext } from 'vscode-azureextensionui';
+
+const deployToAzureExtensionId = 'ms-vscode-deploy-azure.azure-deploy';
+const configurePipelineCommand = 'configure-cicd-pipeline';
+const browsePipelineCommand = 'browse-cicd-pipeline';
+
+export async function configurePipeline(context: IActionContext, target: any): Promise<void> {
+    if (await isDeployToAzureExtensionInstalled()) {
+        await getClusterAndExecute(target, configurePipelineCommand);
+    }
+}
+
+export async function browsePipeline(context: IActionContext, target: any): Promise<void> {
+    if (await isDeployToAzureExtensionInstalled()) {
+        await getClusterAndExecute(target, browsePipelineCommand);
+    }
+}
+
+async function getClusterAndExecute(target: any, command: string): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+    if (cloudExplorer.available) {
+        const clusterTarget = cloudExplorer.api.resolveCommandTarget(target);
+        if (clusterTarget && clusterTarget.cloudName === "Azure" && clusterTarget.nodeType === "resource" && clusterTarget.cloudResource.nodeType === "cluster") {
+            await executeDeployToAzureExtensionInstalled(command, clusterTarget);
+        } else {
+            vscode.window.showInformationMessage('This command only applies to AKS clusters.');
+            return;
+        }
+    } else {
+        return;
+    }
+}
+
+async function isDeployToAzureExtensionInstalled(): Promise<boolean> {
+    const pipelinesExtension = vscode.extensions.getExtension(deployToAzureExtensionId);
+    if (!pipelinesExtension) {
+        vscode.window.showWarningMessage('Please install/enable `Deploy to Azure` extension and start again.');
+        await vscode.commands.executeCommand('extension.open', deployToAzureExtensionId);
+        return false;
+    }
+    return true;
+}
+
+async function executeDeployToAzureExtensionInstalled(commandToRun: string, cluster: any): Promise<void> {
+    const listOfCommands = await vscode.commands.getCommands();
+    if (listOfCommands.find((commmand: string) => commmand === commandToRun)) {
+        return vscode.commands.executeCommand(commandToRun, cluster);
+    }
+    throw new Error(utils.format(`Unable to find command ${commandToRun}. Make sure Deploy to Azure extension is installed and enabled.`));
```

Comment:
💡 `Error(utils.format())` What will this `Error` throw will do for the end-user, i.e. do you guys also need something like `showErrorMessage` to display it to the end user?

Reply:
Is there a top level logic which checks for any error and shows a message to user? If no (which seems to be the case here), then yes we should be showing error to user rather than throw.

Reply:
Just throwing the error gives the error to user  as below
![image](https://user-images.githubusercontent.com/58769601/93876138-f1129200-fcf3-11ea-92a8-d7e08d912085.png)

Reply:
💡 Re: `Error Handling` - I would recommend to error-handle known `error` as extension level `error-handling` i.e. see this sample when you use: `vscode.window.showErrorMessage('Hello World!');` have a play around and possible handle this way because in the event of "firing" and let vscode handle it, it will have the button of "Report an issue" i.e. report this although what I see is that this is known issue for the work in PR and should be handle that way. 

![image](https://user-images.githubusercontent.com/6233295/93948941-d55ec880-fd93-11ea-89bf-dd559ad26f70.png)


Please also feel free to ping me in teams if you want to chatup.

Reply:
> 💡 **Suggestion**: We should handle error with better information, i.e. for example below error which is visible from `AKS Extension` as part of this work, and is not visible in the PR. I checkout the fork just to play around, and I think we should better represent these error at the scope of this extension.
> 
> I think we should clearly handle the errors below when run under this extension.
> 
> * How I produces below error: `right click` on cluster and choose --> `Browse All Pipelines`
> 
> Questions: (Without knowing much about the "deploy to azure" extension.)
> 
> Below question is to build up to the self question I asked myself that: Can we better handle this scenario?
> 
> * What is the scenario which cause folks to directly open the issue against the parent git repository for this issue?
> * Wondering why it is not handled in the extension level with possible docs information as to how to fix it, because with this `open issue` option as a user I have nothing else to report but just the information that I took one random cluster and clicked the `Browse All Pipelines`.
> 
> ![image](https://user-images.githubusercontent.com/6233295/94215775-b5fda200-ff31-11ea-8efb-e824643dcb21.png)
> 
> Thanks heaps.

I guess before we go much into the error handling part , I will explain a bit about how are we trying to implement the fucnctionality.
The menu items which are added in this extension, invoke the commands exposed by Deploy to Azure extension. SO, in parallel, two things are being developed. UI support on this extension as well as backend support in Deploy to Azure etension. The error which you received on clicking on Browse all pipelines is because the browse functionality isn't released yet, and that's why it is breaking the flow.

Now, when it comes to error throwing mechanism., we believe the "Source" part of the error message thrown provides more clarity from which extension the error is coming. This is internal handling by VSCode as we are not doing anything explictly for this. Also, the message that "raise a issue in this repo" is pointing to the repo responsible for this issue. Is there something which you  feel should be done additional, we can surely discuss.

Reply:
* Cool, sounds good. In that case can we please set this PR as Draft PR, because the underlining work which this PR is trying to plumb I.e. “browse all pipeline” is not ready / released yet.

* Yes please I believe regarding error handling we should be as explicit as possible that the source of the error should be clearly underlined rather then invoking the same message which the other extension is using.

A good error handling will not only make the extension control flow clear but also make users workflow cleaner.

Thanks

Reply:
📓 This whole flow is changed to url forwarding. cc: @kanika1894 - Resolving this for and thank you for organising the chat.